src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -24,6 +27,45 @@ internal static partial class Ssl
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslCtxSetClientCertCallback")]
         internal static extern void SslCtxSetClientCertCallback(IntPtr ctx, ClientCertCallback callback);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslCtxSetAlpnProtos")]
+        internal static extern int SslCtxSetAlpnProtos(SafeSslContextHandle ctx, IntPtr protos, int len);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslCtxSetAplnSelectCb")]
+        internal static unsafe extern void SslCtxSetAplnSelectCb(SafeSslContextHandle ctx, SslCtxSetAplnCallback callback, IntPtr arg);
+
+        internal static unsafe int SslCtxSetAplnProtos(SafeSslContextHandle ctx, IList<string> protocols)
+        {
+            byte[] buffer = AlpnStringListToByteArray(protocols);
+            fixed (byte* b = buffer)
+            {
+                return SslCtxSetAlpnProtos(ctx, (IntPtr)b, buffer.Length);
+            }
+        }
+
+        internal static byte[] AlpnStringListToByteArray(IList<string> protocols)
+        {
+            int protocolSize = 0;
+            foreach (string protocol in protocols)
+            {
+                if (string.IsNullOrEmpty(protocol) || protocol.Length > byte.MaxValue)
+                {
+                    throw new ArgumentException(SR.net_ssl_app_protocols_invalid, nameof(protocols));
+                }
+
+                protocolSize += protocol.Length + 1;
+            }
+
+            byte[] buffer = new byte[protocolSize];
+            var offset = 0;
+            foreach (string protocol in protocols)
+            {
+                buffer[offset++] = (byte)(protocol.Length);
+                offset += Encoding.ASCII.GetBytes(protocol, 0, protocol.Length, buffer, offset);
+            }
+
+            return buffer;
+        }
```

Comment:
This function has similar encoding issues, we are now using ASCII encoding, but Protocols struct uses UTF8, and above we are decoding with PtrToAnsi?

Reply:
This code doesn't exist, you're probably looking at an older commit.

Reply:
Correct you updated it after the comment so GH marks it outdated so all good now :thumbsup:

Reply:
Nope, I had initially submitted the PR with this code, this code existed in the first commit, I had then changed this code after the api-proposal to use Readonlymemory in the later commits. You must have looked at this commit wise, instead of the file-changed wise.